lib/src/spec/common.dart

```diff
@@ -261,37 +261,81 @@ class TokenRequest {
 
 }
 
+/// Params for rest history
+///
+/// https://docs.ably.io/client-lib-development-guide/features/#RSL2b
+class RestHistoryParams {
+  /// start must be equal to or less than end and is unaffected by the request direction
+  ///
+  /// RLS2b1
+  final DateTime start;
 
-abstract class RestHistoryParams {
-  int start;
-  int end;
-  String direction;
-  int limit;
+  /// end must be equal to or greater than start and is unaffected by the request direction
+  ///
+  /// RLS2b1
+  final DateTime end;
+
+  /// Sorting history backwards or forwards
+  ///
+  /// if omitted the direction defaults to the REST API default (backwards)
+  /// RLS2b2
+  final String direction;
```

Comment:
The commentary here should specify really clearly what values can be set. Though, surely an enum would be more appropriate? This is a good example of somewhere we could do better in terms of using the compiler.

Reply:
Yes, I too had a thought on this. We can change this to an enum with 2 values and handle it in codec.

https://github.com/ably/ably-flutter/issues/83

Reply:
Commentary was later improved in another dedicated PR for docs #69

Reply:
Perfect, thanks. 😍